test/NonSiloTests/Serialization/BuiltInSerializerTests.cs

```diff
@@ -727,9 +729,9 @@ public void Serialize_Uri_Multithreaded(SerializerToUse serializerToUse)
         [InlineData(SerializerToUse.NoFallback)]
         public void Serialize_GrainReference(SerializerToUse serializerToUse)
         {
-            InitializeSerializer(serializerToUse);
+            var environment = InitializeSerializer(serializerToUse);
             GrainId grainId = GrainId.NewId();
-            GrainReference input = GrainReference.FromGrainId(grainId);
+            GrainReference input = environment.InternalGrainFactory.GetGrain(grainId);
 
             object deserialized = OrleansSerializationLoop(input);
```

Comment:
you probably want to assert that once deserialized, it is bound to the runtime client that is in the context? Or are you asserting that somewhere else?